### PR TITLE
rename Partitioning to Schedule

### PR DESCRIPTION
**Pull request by honnix**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ To define a schedule, simply write a  yaml file to `/etc/styx` (given the above
 ```yaml
 schedules:
   - id: my-workflow
-    partitioning: hours
+    schedule: hours
```
Comment (danielnorberg) [inline README.md]: `s/hours/hourly` ?

File: styx-common/src/main/java/com/spotify/styx/model/Schedule.java [hunk 2]
```diff
@@ -24,8 +24,7 @@
 import com.fasterxml.jackson.annotation.JsonValue;
 import java.util.Locale;
 
-// TODO change class name
-public enum Partitioning {
+public enum Schedule {
   HOURS, DAYS, WEEKS, MONTHS;
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/Schedule.java]: Should these be renamed to `HOURLY`, `DAILY`, `WEEKLY` and `MONTHLY` ?

File: styx-common/src/main/java/com/spotify/styx/util/ParameterUtil.java [hunk 3]
```diff
@@ -181,21 +181,21 @@ public static Instant truncateInstant(Instant instant, Partitioning partitioning
...
       default:
-        throw new IllegalArgumentException("Partitioning not supported: " + partitioning);
+        throw new IllegalArgumentException("Schedule not supported: " + schedule);
     }
   }
 
   /**
-   * Generates a list of {@link Instant}s obtained by partitioning a time range defined by a
-   * starting and a ending {@link Instant}s, and based on the provided {@link Partitioning}.
+   * Generates a list of {@link Instant}s obtained by schedule a time range defined by a
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/util/ParameterUtil.java]: This sentence needs fixing

Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/util/ParameterUtil.java]: they will be fixed.

File: styx-common/src/main/java/com/spotify/styx/model/Schedule.java (hunk 2 above)
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/Schedule.java]: not sure of the history here actually.

File: styx-common/src/main/java/com/spotify/styx/util/ParameterUtil.java [hunk 4]
```diff
@@ -181,21 +181,21 @@ public static Instant truncateInstant(Instant instant, Partitioning partitioning
...
+   * starting and a ending {@link Instant}s, and based on the provided {@link Schedule}.
    *
    * @param startInstant              Defines the start of the time range (inclusive)
    * @param endInstant                Defines the end of the time range (exclusive)
-   * @param partitioning              The partitioning unit to split the time range into
+   * @param schedule              The schedule unit to split the time range into
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/util/ParameterUtil.java]: Indentation

File: styx-common/src/main/java/com/spotify/styx/model/Schedule.java (hunk 2 above)
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/Schedule.java]: it looks we prefer `.s` over `.ly`

Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/Schedule.java]: How about we keep it unchanged for now?

Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/Schedule.java]: 👍

**Comment by danielnorberg** [comment]
👍

**Comment by rouzwawi** [comment]
It would have been nice to do this rename after merging the cron branch. Got a pretty huge conflict to resolve now...

**Comment by honnix** [comment]
Sorry I realized this after we merged and released. Would it help or possible to rebase from the version before the change first and then to the head?

**Comment by rouzwawi** [comment]
I'm trying to work through resolving the conflicts. Will try that if this take too long.

**Pull request merged**